pennylane/compiler/qjit_api.py

```diff
@@ -460,18 +461,22 @@ def _call_capture_enabled(self, *init_state):
 
         while_loop_prim = _get_while_loop_qfunc_prim()
 
-        jaxpr_body_fn = jax.make_jaxpr(self.body_fn)(*init_state)
+        flat_body_fn = FlatFn(self.body_fn)
+        jaxpr_body_fn = jax.make_jaxpr(flat_body_fn)(*init_state)
         jaxpr_cond_fn = jax.make_jaxpr(self.cond_fn)(*init_state)
```

Comment:
We don't need `pytree` support for `cond_fn`?

Reply:
`FlatFn` serves to capture the output pytree shape so we can repack the results at the end. `cond_fn` must purely return a boolean. We don't care about the pytree shape of the output, and it wouldn't effect anything we do.  We just care that the only leaf is a boolean.